File: 250827/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.py
def is_prime(num):
    for i in range(2, num):
        if num%i == 0:
            return False
    return True

def count(A, B):
    # 소수이면서
    # 모든 자릿수의 합이 짝수인 수의 개수
    # 1~100
    cnt = 0

    for num in range(A, B+1):
        # 소수가 아니면 다음 숫자로 넘어감
        if not is_prime(num):
            continue
        
        if (num//10 + num%10)%2 == 0:
            cnt += 1
    
    return cnt

A, B = map(int, input().split())
print(count(A, B))

File: trail2/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();

        int cnt = 0;
        for (int i=a; i<=b; i++) {
            if (isPrime(i) && isEvenSum(i)) cnt++;
        }
        System.out.println(cnt);
    }

    public static boolean isPrime(int n) {
        for (int i=2; i<n; i++) {
            if (n%i==0) return false;
        }
        return true;
    }

    public static boolean isEvenSum(int n) {
        int sum = 0;
        int num = n;
        while (num>0) {
            sum += num%10;
            num = num/10;
        }

        if (sum%2==0) return true;
        return false;
    }
}